pkg/services/dashboards/service/dashboard_service.go

```diff
@@ -318,10 +318,7 @@ func (dr *DashboardServiceImpl) SaveProvisionedDashboard(ctx context.Context, dt
 	}
 
 	if dto.Dashboard.ID == 0 {
-		if err := dr.setDefaultPermissions(ctx, dto, dash, true); err != nil {
-			namespaceID, userID := dto.User.GetNamespacedID()
-			dr.log.Error("Could not make user admin", "dashboard", dash.Title, "namespaceID", namespaceID, "userID", userID, "error", err)
```

Comment:
The main change is in [here](https://github.com/grafana/grafana/pull/77155/files#diff-56c23b161208f11acd3ddd098e59329339a548161430a3d7303a1e80e5ed2c8fR479-R491) - previously we tried to get `userID` before checking if the dashboard is being created by provisioning. Because provisioning isn't a real user, we failed to get int identifier [here](https://github.com/grafana/grafana/pull/77155/files#diff-56c23b161208f11acd3ddd098e59329339a548161430a3d7303a1e80e5ed2c8fL494-L496), and no permissions were set for the dashboard. Now we check if the dashboard is being created by provisioning before we attempt to get `userID`, so we don't fail there and permissions are correctly set for the basic roles.

Reply:
thx for pointing it out. missed that err check when reviewing 😵‍💫